File: src/main/java/eu/codetopic/utils/ui/activity/modular/module/ToolbarModule.java
/*
 * Copyright 2017 Jiří Kuchyňka (Anty)
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */

package eu.codetopic.utils.ui.activity.modular.module;

import android.support.annotation.LayoutRes;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.view.ViewGroup;

import eu.codetopic.utils.R;
import eu.codetopic.utils.ui.activity.modular.ModularActivity;
import eu.codetopic.utils.ui.activity.modular.SimpleActivityCallBackModule;

public class ToolbarModule extends SimpleActivityCallBackModule {

    private static final String LOG_TAG = "ToolbarModule";

    @Override
    protected void onSetContentView(@LayoutRes final int layoutResID, SetContentViewCallBack callBack) {
        callBack.set(R.layout.toolbar_base);
        callBack.addViewAttachedCallBack(() -> {
            ModularActivity activity = getActivity();
            activity.getLayoutInflater().inflate(layoutResID, activity.findViewById(R.id.base_content));
        });
        setupCallback(callBack);
    }

    @Override
    protected void onSetContentView(final View view, SetContentViewCallBack callBack) {
        callBack.set(R.layout.toolbar_base);
        callBack.addViewAttachedCallBack(() -> {
            ((ViewGroup) getActivity().findViewById(R.id.base_content)).addView(view);
        });
        setupCallback(callBack);
    }

    @Override
    protected void onSetContentView(final View view, final ViewGroup.LayoutParams params, SetContentViewCallBack callBack) {
        callBack.set(R.layout.toolbar_base);
        callBack.addViewAttachedCallBack(() -> {
            ((ViewGroup) getActivity().findViewById(R.id.base_content)).addView(view, params);
        });
        setupCallback(callBack);
    }

    private void setupCallback(SetContentViewCallBack callBack) {
        callBack.addViewAttachedCallBack(() -> {
            ModularActivity activity = getActivity();
            activity.setSupportActionBar(activity.findViewById(R.id.toolbar));
        });
    }
}

File: src/main/java/eu/codetopic/utils/ui/activity/modular/module/ToolbarModule.kt
/*
 * utils
 * Copyright (C)   2017  anty
 *
 * This program is free  software: you can redistribute it and/or modify
 * it under the terms  of the GNU General Public License as published by
 * the Free Software  Foundation, either version 3 of the License, or
 * (at your option) any  later version.
 *
 * This program is distributed in the hope that it  will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied  warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
 * GNU General Public License for more details.
 *
 * You  should have received a copy of the GNU General Public License
 * along  with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package eu.codetopic.utils.ui.activity.modular.module

import android.support.annotation.LayoutRes
import android.support.v7.widget.Toolbar
import android.view.View
import android.view.ViewGroup

import eu.codetopic.utils.R
import eu.codetopic.utils.ui.activity.modular.ActivityCallBackModule
import eu.codetopic.utils.ui.activity.modular.ActivityCallBackModule.SetContentViewCallBack
import eu.codetopic.utils.ui.activity.modular.ModularActivity
import eu.codetopic.utils.ui.activity.modular.SimpleActivityCallBackModule

class ToolbarModule : SimpleActivityCallBackModule() {

    companion object {

        private const val LOG_TAG = "ToolbarModule"
    }

    override fun onSetContentView(@LayoutRes layoutResID: Int, callBack: SetContentViewCallBack) {
        callBack.set(R.layout.toolbar_base)
        callBack.addViewAttachedCallBack {
            activity.let {
                it.layoutInflater.inflate(layoutResID, it.findViewById(R.id.base_content))
            }
        }
        setupCallback(callBack)
    }

    override fun onSetContentView(view: View, callBack: SetContentViewCallBack) {
        callBack.set(R.layout.toolbar_base)
        callBack.addViewAttachedCallBack {
            activity.findViewById<ViewGroup>(R.id.base_content).addView(view)
        }
        setupCallback(callBack)
    }

    override fun onSetContentView(view: View, params: ViewGroup.LayoutParams, callBack: SetContentViewCallBack) {
        callBack.set(R.layout.toolbar_base)
        callBack.addViewAttachedCallBack {
            (activity.findViewById<View>(R.id.base_content) as ViewGroup).addView(view, params)
        }
        setupCallback(callBack)
    }

    private fun setupCallback(callBack: SetContentViewCallBack) {
        callBack.addViewAttachedCallBack({
            activity.let { it.setSupportActionBar(it.findViewById(R.id.toolbar)) }
        })
    }
}
